corehq/apps/sms/phonenumbers_helper.py

```diff
@@ -17,23 +17,44 @@ def parse_phone_number(number, region=None, failhard=True):
     from phonenumbers.phonenumberutil import parse as phonenumbers_parse, NumberParseException
     try:
         phone = phonenumbers_parse(number, region)
-        if phone:
-            return PhoneNumber(phone.country_code, phone.national_number)
     except NumberParseException:
         if failhard:
             raise PhoneNumberParseException()
         else:
             return None
 
+    return PhoneNumber(phone.country_code, phone.national_number)
+
 
 def strip_plus(number):
     return number[1:] if number.startswith('+') else number
 
 
-def get_country_code_and_national_number(number, failhard=False):
-    parsed = parse_phone_number(number, failhard=failhard)
-    if parsed:
-        country_code = parsed.country_code
-        national_number = strip_plus(number)[len(str(country_code)):]
-        return country_code, national_number
-    return None, None
+def get_country_code_and_national_number(number):
+    """
+    :param number: str representing phone number
+    :return: country_code: int, national_number: str
```

Comment:
These types listed here are incomplete since this may return `None, None`
```suggestion
    :return: country_code: int, national_number: str or, if the number
    could not be parsed, `None, None`.
```